esp-hal/src/timer/mod.rs

```diff
@@ -105,22 +104,40 @@ pub trait Timer: crate::private::Sealed {
     /// Has the timer triggered?
     fn is_interrupt_set(&self) -> bool;
 
+    /// Asynchronously wait for the timer interrupt to fire.
+    ///
+    /// Requires the correct `InterruptHandler` to be installed to function
+    /// correctly.
+    async fn wait(&self);
```

Comment:
I tried an API where it just returned a future, but it was noising and I had trouble getting delegate to play ball with the types.

I think this should be fine, but let me know.